
Mageia support is partly added upstream beginning with 7.0 branch.

This patch adds the missing bits to extend / improve the integration.

Signed-off-by: Thomas Backlund <tmb@mageia.org>

---
 src/VBox/Additions/linux/installer/vboxadd-x11.sh        |    6 ++++++
 src/VBox/Frontends/VirtualBox/VirtualBox2.qrc            |    2 ++
 src/VBox/Frontends/VirtualBox/VirtualBox2_x2.qrc         |    2 ++
 src/VBox/Frontends/VirtualBox/VirtualBox2_x3.qrc         |    2 ++
 src/VBox/Frontends/VirtualBox/VirtualBox2_x4.qrc         |    2 ++
 src/VBox/Frontends/VirtualBox/src/globals/UIIconPool.cpp |    4 ++--
 src/VBox/Installer/linux/VBoxSysInfo.sh                  |    5 +++++
 src/VBox/Installer/linux/distributions_rpm               |   15 +++++++++++++++
 src/VBox/Main/src-server/ApplianceImpl.cpp               |    2 ++
 src/VBox/Main/src-server/UnattendedImpl.cpp              |    6 ++++++
 src/VBox/Main/xml/Settings.cpp                           |    1 +
 11 files changed, 45 insertions(+), 2 deletions(-)

diff -Nurp VirtualBox-7.0.2.orig/src/VBox/Additions/linux/installer/vboxadd-x11.sh VirtualBox-7.0.2/src/VBox/Additions/linux/installer/vboxadd-x11.sh
--- VirtualBox-7.0.2.orig/src/VBox/Additions/linux/installer/vboxadd-x11.sh	2022-10-19 21:20:43.000000000 +0300
+++ VirtualBox-7.0.2/src/VBox/Additions/linux/installer/vboxadd-x11.sh	2022-10-22 12:34:07.913136643 +0300
@@ -104,8 +104,14 @@ install_x11_startup_app() {
     x11_autostart="/etc/xdg/autostart"
     kde_autostart="/usr/share/autostart"
     redhat_dir=/etc/X11/Xsession.d
+    mageia_dir=/etc/X11/xinit.d
     mandriva_dir=/etc/X11/xinit.d
     debian_dir=/etc/X11/xinit/xinitrc.d
+    if [ -d "$mageia_dir" -a -w "$mageia_dir" -a -x "$mageia_dir" ]
+    then
+        install -m 0644 $app_src "$mageia_dir/$app_dest"
+        found=1
+    fi
     if [ -d "$mandriva_dir" -a -w "$mandriva_dir" -a -x "$mandriva_dir" ]
     then
         install -m 0644 $app_src "$mandriva_dir/$app_dest"
diff -Nurp VirtualBox-7.0.2.orig/src/VBox/Frontends/VirtualBox/src/globals/UIIconPool.cpp VirtualBox-7.0.2/src/VBox/Frontends/VirtualBox/src/globals/UIIconPool.cpp
--- VirtualBox-7.0.2.orig/src/VBox/Frontends/VirtualBox/src/globals/UIIconPool.cpp	2022-10-19 21:28:31.000000000 +0300
+++ VirtualBox-7.0.2/src/VBox/Frontends/VirtualBox/src/globals/UIIconPool.cpp	2022-10-22 12:44:34.937117962 +0300
@@ -416,8 +416,8 @@ UIIconPoolGeneral::UIIconPoolGeneral()
     m_guestOSTypeIconNames.insert("OpenMandriva_Lx_64", ":/os_mandriva_64.png");
     m_guestOSTypeIconNames.insert("PCLinuxOS",       ":/os_mandriva.png");
     m_guestOSTypeIconNames.insert("PCLinuxOS_64",    ":/os_mandriva_64.png");
-    m_guestOSTypeIconNames.insert("Mageia",          ":/os_mandriva.png");
-    m_guestOSTypeIconNames.insert("Mageia_64",       ":/os_mandriva_64.png");
+    m_guestOSTypeIconNames.insert("Mageia",          ":/os_mageia.png");
+    m_guestOSTypeIconNames.insert("Mageia_64",       ":/os_mageia_64.png");
     m_guestOSTypeIconNames.insert("RedHat",          ":/os_redhat.png");
     m_guestOSTypeIconNames.insert("RedHat_64",       ":/os_redhat_64.png");
     m_guestOSTypeIconNames.insert("RedHat3",         ":/os_redhat.png");
diff -Nurp VirtualBox-7.0.2.orig/src/VBox/Frontends/VirtualBox/VirtualBox2.qrc VirtualBox-7.0.2/src/VBox/Frontends/VirtualBox/VirtualBox2.qrc
--- VirtualBox-7.0.2.orig/src/VBox/Frontends/VirtualBox/VirtualBox2.qrc	2022-10-19 21:28:04.000000000 +0300
+++ VirtualBox-7.0.2/src/VBox/Frontends/VirtualBox/VirtualBox2.qrc	2022-10-22 12:39:06.604127744 +0300
@@ -92,6 +92,8 @@
         <file alias="os_linux.png">images/os_linux.png</file>
         <file alias="os_macosx_64.png">images/os_macosx_64.png</file>
         <file alias="os_macosx.png">images/os_macosx.png</file>
+        <file alias="os_mageia_64.png">images/os_mageia_64.png</file>
+        <file alias="os_mageia.png">images/os_mageia.png</file>
         <file alias="os_mandriva_64.png">images/os_mandriva_64.png</file>
         <file alias="os_mandriva.png">images/os_mandriva.png</file>
         <file alias="os_netbsd_64.png">images/os_netbsd_64.png</file>
diff -Nurp VirtualBox-7.0.2.orig/src/VBox/Frontends/VirtualBox/VirtualBox2_x2.qrc VirtualBox-7.0.2/src/VBox/Frontends/VirtualBox/VirtualBox2_x2.qrc
--- VirtualBox-7.0.2.orig/src/VBox/Frontends/VirtualBox/VirtualBox2_x2.qrc	2022-10-19 21:28:04.000000000 +0300
+++ VirtualBox-7.0.2/src/VBox/Frontends/VirtualBox/VirtualBox2_x2.qrc	2022-10-22 12:37:55.554129861 +0300
@@ -103,6 +103,8 @@
         <file alias="os_linux_x2.png">images/x2/os_linux_x2.png</file>
         <file alias="os_macosx_64_x2.png">images/x2/os_macosx_64_x2.png</file>
         <file alias="os_macosx_x2.png">images/x2/os_macosx_x2.png</file>
+        <file alias="os_mageia_64_x2.png">images/x2/os_mageia_64_x2.png</file>
+        <file alias="os_mageia_x2.png">images/x2/os_mageia_x2.png</file>
         <file alias="os_mandriva_64_x2.png">images/x2/os_mandriva_64_x2.png</file>
         <file alias="os_mandriva_x2.png">images/x2/os_mandriva_x2.png</file>
         <file alias="os_netbsd_64_x2.png">images/x2/os_netbsd_64_x2.png</file>
diff -Nurp VirtualBox-7.0.2.orig/src/VBox/Frontends/VirtualBox/VirtualBox2_x3.qrc VirtualBox-7.0.2/src/VBox/Frontends/VirtualBox/VirtualBox2_x3.qrc
--- VirtualBox-7.0.2.orig/src/VBox/Frontends/VirtualBox/VirtualBox2_x3.qrc	2022-10-19 21:28:04.000000000 +0300
+++ VirtualBox-7.0.2/src/VBox/Frontends/VirtualBox/VirtualBox2_x3.qrc	2022-10-22 12:38:02.478129654 +0300
@@ -103,6 +103,8 @@
         <file alias="os_linux_x3.png">images/x3/os_linux_x3.png</file>
         <file alias="os_macosx_64_x3.png">images/x3/os_macosx_64_x3.png</file>
         <file alias="os_macosx_x3.png">images/x3/os_macosx_x3.png</file>
+        <file alias="os_mageia_64_x3.png">images/x3/os_mageia_64_x3.png</file>
+        <file alias="os_mageia_x3.png">images/x3/os_mageia_x3.png</file>
         <file alias="os_mandriva_64_x3.png">images/x3/os_mandriva_64_x3.png</file>
         <file alias="os_mandriva_x3.png">images/x3/os_mandriva_x3.png</file>
         <file alias="os_netbsd_64_x3.png">images/x3/os_netbsd_64_x3.png</file>
diff -Nurp VirtualBox-7.0.2.orig/src/VBox/Frontends/VirtualBox/VirtualBox2_x4.qrc VirtualBox-7.0.2/src/VBox/Frontends/VirtualBox/VirtualBox2_x4.qrc
--- VirtualBox-7.0.2.orig/src/VBox/Frontends/VirtualBox/VirtualBox2_x4.qrc	2022-10-19 21:28:04.000000000 +0300
+++ VirtualBox-7.0.2/src/VBox/Frontends/VirtualBox/VirtualBox2_x4.qrc	2022-10-22 12:38:08.986129460 +0300
@@ -103,6 +103,8 @@
         <file alias="os_linux_x4.png">images/x4/os_linux_x4.png</file>
         <file alias="os_macosx_64_x4.png">images/x4/os_macosx_64_x4.png</file>
         <file alias="os_macosx_x4.png">images/x4/os_macosx_x4.png</file>
+        <file alias="os_mageia_64_x4.png">images/x4/os_mageia_64_x4.png</file>
+        <file alias="os_mageia_x4.png">images/x4/os_mageia_x4.png</file>
         <file alias="os_mandriva_64_x4.png">images/x4/os_mandriva_64_x4.png</file>
         <file alias="os_mandriva_x4.png">images/x4/os_mandriva_x4.png</file>
         <file alias="os_netbsd_64_x4.png">images/x4/os_netbsd_64_x4.png</file>
diff -Nurp VirtualBox-7.0.2.orig/src/VBox/Installer/linux/distributions_rpm VirtualBox-7.0.2/src/VBox/Installer/linux/distributions_rpm
--- VirtualBox-7.0.2.orig/src/VBox/Installer/linux/distributions_rpm	2022-10-19 21:28:54.000000000 +0300
+++ VirtualBox-7.0.2/src/VBox/Installer/linux/distributions_rpm	2022-10-22 12:48:25.951111079 +0300
@@ -2,6 +2,21 @@ openSUSETW      = OPENSUSE_TW
 openSUSE155     = OPENSUSE_15_5
 openSUSE154     = OPENSUSE_15_4
 openSUSE153     = OPENSUSE_15_3
+mga15           = MAGEIA_15
+mga14           = MAGEIA_14
+mga13           = MAGEIA_13
+mga12           = MAGEIA_12
+mga11           = MAGEIA_11
+mga10           = MAGEIA_10
+mga9            = MAGEIA_9
+mga8            = MAGEIA_8
+mga7            = MAGEIA_7
+mga6            = MAGEIA_6
+mga5            = MAGEIA_5
+mga4            = MAGEIA_4
+mga3            = MAGEIA_3
+mga2            = MAGEIA_2
+mga1            = MAGEIA_1
 fedora40        = FEDORA_40
 fedora38        = FEDORA_38
 fedora37        = FEDORA_37
diff -Nurp VirtualBox-7.0.2.orig/src/VBox/Installer/linux/VBoxSysInfo.sh VirtualBox-7.0.2/src/VBox/Installer/linux/VBoxSysInfo.sh
--- VirtualBox-7.0.2.orig/src/VBox/Installer/linux/VBoxSysInfo.sh	2022-10-19 21:28:54.000000000 +0300
+++ VirtualBox-7.0.2/src/VBox/Installer/linux/VBoxSysInfo.sh	2022-10-22 12:48:15.439111393 +0300
@@ -54,6 +54,11 @@ get_linux_info () {
         # Debian-based system
         release=`cat /etc/debian_version`
         print_linux_info "Debian" $release
+    elif [ -r /etc/mageia-release ]
+    then
+        # Mageia-based system
+        release=`cat /etc/mageia-release | sed -e 's/[A-Za-z ]* release //'`
+        print_linux_info "Mageia" $release
     elif [ -r /etc/mandriva-release ]
     then
         # Mandriva-based system
diff -Nurp VirtualBox-7.0.2.orig/src/VBox/Main/src-server/ApplianceImpl.cpp VirtualBox-7.0.2/src/VBox/Main/src-server/ApplianceImpl.cpp
--- VirtualBox-7.0.2.orig/src/VBox/Main/src-server/ApplianceImpl.cpp	2022-10-19 21:29:03.000000000 +0300
+++ VirtualBox-7.0.2/src/VBox/Main/src-server/ApplianceImpl.cpp	2022-10-22 12:57:29.027094900 +0300
@@ -316,6 +316,7 @@ static const osTypePattern g_aOsTypesPat
     {"SUSE",          VBOXOSTYPE_OpenSUSE},
     {"Novell",        VBOXOSTYPE_OpenSUSE},
     {"Red Hat",       VBOXOSTYPE_RedHat},
+    {"Mageia",        VBOXOSTYPE_Mageia},
     {"Mandriva",      VBOXOSTYPE_Mandriva},
     {"Ubuntu",        VBOXOSTYPE_Ubuntu},
     {"Debian",        VBOXOSTYPE_Debian},
@@ -352,6 +353,7 @@ static const osTypePattern g_aOsTypesPat
     {"SUSE",          VBOXOSTYPE_OpenSUSE_x64},
     {"Novell",        VBOXOSTYPE_OpenSUSE_x64},
     {"Red Hat",       VBOXOSTYPE_RedHat_x64},
+    {"Mageia",        VBOXOSTYPE_Mageia_x64},
     {"Mandriva",      VBOXOSTYPE_Mandriva_x64},
     {"Ubuntu",        VBOXOSTYPE_Ubuntu_x64},
     {"Debian",        VBOXOSTYPE_Debian_x64},
diff -Nurp VirtualBox-7.0.2.orig/src/VBox/Main/src-server/UnattendedImpl.cpp VirtualBox-7.0.2/src/VBox/Main/src-server/UnattendedImpl.cpp
--- VirtualBox-7.0.2.orig/src/VBox/Main/src-server/UnattendedImpl.cpp	2022-10-19 21:29:04.000000000 +0300
+++ VirtualBox-7.0.2/src/VBox/Main/src-server/UnattendedImpl.cpp	2022-10-22 13:03:28.835084180 +0300
@@ -1394,6 +1394,12 @@ static bool detectLinuxDistroName(const
         *penmOsType = (VBOXOSTYPE)((*penmOsType & VBOXOSTYPE_ArchitectureMask) | VBOXOSTYPE_Oracle);
         pszOsAndVersion = RTStrStripL(pszOsAndVersion + 6);
     }
+    else if (   RTStrNICmp(pszOsAndVersion, RT_STR_TUPLE("Mageia")) == 0
+             && !RT_C_IS_ALNUM(pszOsAndVersion[6]))
+    {
+        *penmOsType = (VBOXOSTYPE)((*penmOsType & VBOXOSTYPE_ArchitectureMask) | VBOXOSTYPE_Mageia);
+        pszOsAndVersion = RTStrStripL(pszOsAndVersion + 6);
+    }
     else if (   RTStrNICmp(pszOsAndVersion, RT_STR_TUPLE("CentOS")) == 0
              && !RT_C_IS_ALNUM(pszOsAndVersion[6]))
     {
diff -Nurp VirtualBox-7.0.2.orig/src/VBox/Main/xml/Settings.cpp VirtualBox-7.0.2/src/VBox/Main/xml/Settings.cpp
--- VirtualBox-7.0.2.orig/src/VBox/Main/xml/Settings.cpp	2022-10-19 21:29:07.000000000 +0300
+++ VirtualBox-7.0.2/src/VBox/Main/xml/Settings.cpp	2022-10-22 13:04:47.873081825 +0300
@@ -6579,6 +6579,7 @@ const struct {
     { "opensuse", "OpenSUSE" },
     { "fedoracore", "Fedora" },
     { "gentoo", "Gentoo" },
+    { "mageia", "Mageia" },
     { "mandriva", "Mandriva" },
     { "redhat", "RedHat" },
     { "ubuntu", "Ubuntu" },
